File: libs/parse/tests/fixtures/go/hello.go
// Minimal Go fixture used by libs/parse tests. Covers package, import,
// struct, method, interface, goroutine, and channel — the grammar paths
// most Go code we'll scan exercises.

package main

import (
	"context"
	"fmt"
	"sync"
)

type Greeting struct {
	Subject string
	Reps    int
}

type Renderable interface {
	Render() string
}

func (g Greeting) Render() string {
	return fmt.Sprintf("Hello, %s!", g.Subject)
}

func add(a, b int) int {
	return a + b
}

func main() {
	ctx, cancel := context.WithCancel(context.Background())
	defer cancel()

	var wg sync.WaitGroup
	wg.Add(1)
	go func() {
		defer wg.Done()
		g := Greeting{Subject: "world", Reps: add(40, 2)}
		fmt.Println(g.Render(), ctx.Err())
	}()
	wg.Wait()
}

File: libs/parse/tests/fixtures/rust/hello.rs
// Minimal Rust fixture used by libs/parse tests. Exercises function
// definitions, struct types, impl blocks, generic constraints, and
// the `Result` enum — the grammar paths most rust code we'll scan
// goes through.

use std::fmt::Display;

#[derive(Debug, Clone)]
struct Greeting<T: Display> {
    subject: T,
    reps: u32,
}

impl<T: Display> Greeting<T> {
    fn new(subject: T) -> Self {
        Self { subject, reps: 1 }
    }

    fn render(&self) -> String {
        format!("Hello, {}!", self.subject)
    }
}

fn add(a: i32, b: i32) -> i32 {
    a + b
}

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let sum = add(40, 2);
    let g = Greeting::new("world");
    println!("{} ({})", g.render(), sum);
    Ok(())
}
